ethereumj-core/src/main/java/org/ethereum/casper/manager/CasperWorldManager.java

```diff
@@ -0,0 +1,50 @@
+package org.ethereum.casper.manager;
+
+import org.ethereum.casper.config.CasperProperties;
+import org.ethereum.casper.core.CasperFacade;
+import org.ethereum.casper.service.CasperValidatorService;
+import org.ethereum.config.SystemProperties;
+import org.ethereum.core.Blockchain;
+import org.ethereum.core.Genesis;
+import org.ethereum.core.Repository;
+import org.ethereum.casper.core.genesis.CasperStateInit;
+import org.ethereum.core.genesis.StateInit;
+import org.ethereum.manager.WorldManager;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.PostConstruct;
+
+public class CasperWorldManager extends WorldManager {
+
+    @Autowired
+    private CasperValidatorService casperValidatorService;
+
+    @Autowired
+    private CasperFacade casper;
+
+    private CasperStateInit stateInit;
+
+    public CasperWorldManager(SystemProperties config, Repository repository, Blockchain blockchain) {
+        super(config, repository, blockchain);
+    }
+
+    @PostConstruct
+    @Override
+    protected void init() {
+        super.init();
+        casper.setEthereum(ethereum);
+        if (Boolean.TRUE.equals(((CasperProperties) config).getCasperValidatorEnabled())) {
```

Comment:
Just add `@Autowired public void setConfig(SystemProperties config) { ... }` method to 
`CasperValidatorService` like it's done in other similar places

Reply:
We'll fail if this action starts before postconstruct, during Autowired process, because not all dependencies are in place.

Reply:
ok, got it. then maybe it's better to wrap this check in `CasperValidatorService#init()` and just call `init()` here?

Reply:
Ok, looks reasonable